### ISSUE TITLE
QBatchNormalization with scale=False and model_save_quantized_weights

### ISSUE THREAD

**Issue by lattuada-st** [google/qkeras#83]
When model_save_quantized_weights is called on a model including a QBatchNormalization with scale=False it seems that the wrong quantizers are used.
QBatchNormalization.get_quantizers() returns a list with gamma_quantizer as first element even when there is no gamma, resulting in a disalignment between quantizers and weights in this point https://github.com/google/qkeras/blob/1f2134b48548a548f22ee7b75079cb9e34eaff5b/qkeras/utils.py#L159

**Comment by danielemoro** [comment]
Hello. Thank you for reporting this.

Can you please provide a code example where I can reproduce the problem?

**Comment by danielemoro** [comment]
So if I understand correctly, the issue is caused by this line: https://github.com/google/qkeras/blob/1f2134b48548a548f22ee7b75079cb9e34eaff5b/qkeras/utils.py#L159

```
for quantizer, weight in zip(qs, ws):
```

If `len(qs) != len(ws)`, then the wrong quantizer will be used to quantize each weight.

There may be several solutions to this, but I will need some time to investigate

**Comment by lattuada-st** [comment]
Yes, the issue is there

**Comment by danielemoro** [comment]
We have determined that to fix this issue will take quite a bit of refactoring work, and so it will take more time. 

In the meantime, the workaround is to use [folded batch normalization](https://towardsdatascience.com/speed-up-inference-with-batch-normalization-folding-8a45a83a89d8). This method allows you to integrate your batch normalization operations into your convolution layers, thereby avoiding batch norm layers altogether. This is the method we currently use in our team.

We have good support for this in QKeras if you switch to using the QConv2DBatchnorm layers. Then you can use the `convert_folded_model_to_normal_seq` function to map a model full of QConv2DBatchnorm layers to an equivalent model with just QConv2D layers. See more info here: https://github.com/google/qkeras/commit/bc618905cc1e4866e787e5c04f3465ea09b5db0e

Let me know if this helps.
